x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/X-BigArrayVector.java.st

```diff
@@ -55,19 +55,22 @@ public final class $Type$BigArrayVector extends AbstractVector implements $Type$
     @Override
     public $Type$Vector filter(int... positions) {
         var blockFactory = blockFactory();
-   $if(boolean)$
+    $if(boolean)$
         final BitArray filtered = new BitArray(positions.length, blockFactory.bigArrays());
+    $else$
+        final $Array$ filtered = blockFactory.bigArrays().new$Array$(positions.length);
+    $endif$
+    $if(boolean)$
         for (int i = 0; i < positions.length; i++) {
             if (values.get(positions[i])) {
                 filtered.set(i);
             }
         }
-   $else$
-        final $Type$Array filtered = blockFactory.bigArrays().new$Type$Array(positions.length, true);
+    $else$
         for (int i = 0; i < positions.length; i++) {
             filtered.set(i, values.get(positions[i]));
         }
-   $endif$
+    $endif$
```

Comment:
Thanks.

Reply:
Instead of just reverting all the generated code stuff I did, I removed all the PointBlock stuff, and then re-reviewed what was left and kept what I thought still seemed like an improvement, even if not necessary for this PR.